cyraft/node.py

```diff
@@ -661,11 +696,19 @@ def _reset_election_timeout(self) -> None:
         assert self._state == RaftState.FOLLOWER, "Only followers should reset the election timeout"
         _logger.info(c["raft_logic"] + "Node ID: %d -- Resetting election timeout" + c["end_color"], self._node.id)
         loop = asyncio.get_event_loop()
-        self._next_election_timeout = loop.time() + self._election_timeout
+        """
+        When on_election_timeout is called, the state is changed to candidate and a new election is started by calling start_election function. 
+        In the start_election function, the election timeout is reset using self._next_election_timeout = loop.time() + self._election_timeout. 
+        If loop.time() returns a value greater than or equal to self._next_election_timeout, then the election timeout will be set to a time in the past, 
+        causing the timeout to trigger immediately.
+        To fix this issue, added a small offset to the current time when resetting the election timeout
+        """
+        self._next_election_timeout = loop.time() + self._election_timeout + 0.01
         if hasattr(self, "_election_timer"):
+            _logger.info(c["raft_logic"] + "Node ID: %d -- Canceling the alredy existed election timer" + c["end_color"], self._node.id)
             self._election_timer.cancel()
-        self._election_timer = loop.call_at(
-            self._next_election_timeout, lambda: asyncio.create_task(self._on_election_timeout())
+        self._election_timer = loop.call_later(
+            self._election_timeout, lambda: asyncio.create_task(self._on_election_timeout())
```

Comment:
Look what I wrote in the comments:
**When on_election_timeout is called, the state is changed to candidate and a new election is started by calling start_election function. In the start_election function, the election timeout is reset using self._next_election_timeout = loop.time() self._election_timeout.  If loop.time() returns a value greater than or equal to self._next_election_timeout, then the election timeout will be set to a time in the past, causing the timeout to trigger immediately. To fix this issue, added a small offset to the current time when resetting the election timeout**
From the next commit I will remove this expression `self._next_election_timeout = loop.time() + self._election_timeout + 0.01` because the `loop.call_later` function is more suitable than `loop.call_at`. It does not need to calculate the next timeout as for `loop.call_at`.